Strip the Qt metatype tails from otherwise Qt-free core headers.
dive.h, divesite.h, trip.h and triptable.h each end with an #include <QObject>
plus a Q_DECLARE_METATYPE for the desktop app's QVariant plumbing. Nothing in
the vendored subset needs it, and it is the only Qt reference in those headers.

--- a/core/dive.h	2026-08-11 20:15:14
+++ b/core/dive.h	2026-08-11 20:15:27
@@ -210,8 +210,6 @@
 /* Make pointers to dive and dive_trip "Qt metatypes" so that they can be passed through
  * QVariants and through QML.
  */
-#include <QObject>
-Q_DECLARE_METATYPE(struct dive *);
 
 extern std::string existing_filename;
 
--- a/core/divesite.h	2026-08-11 20:15:14
+++ b/core/divesite.h	2026-08-11 20:15:27
@@ -35,7 +35,5 @@
 int divesite_comp_uuid(const dive_site &ds1, const dive_site &ds2);
 
 /* Make pointer-to-dive_site a "Qt metatype" so that we can pass it through QVariants */
-#include <QObject>
-Q_DECLARE_METATYPE(dive_site *);
 
 #endif // DIVESITE_H
--- a/core/trip.h	2026-08-11 20:15:14
+++ b/core/trip.h	2026-08-11 20:15:27
@@ -49,7 +49,5 @@
 
 /* Make pointers to dive_trip "Qt metatypes" so that they can be
  * passed through QVariants and through QML. See comment in dive.h. */
-#include <QObject>
-Q_DECLARE_METATYPE(struct dive_trip *);
 
 #endif
--- a/core/triptable.h	2026-08-11 20:22:51
+++ b/core/triptable.h	2026-08-11 20:22:51
@@ -18,8 +18,6 @@
 
 /* Make pointers to trip_table "Qt metatypes" so that they can be
  * passed through QVariants and through QML. See comment in dive.h. */
-#include <QObject>
-Q_DECLARE_METATYPE(trip_table *);
 
 #endif
 
